src/activities/Elsa.Activities.Rpa.Web/Activities/WebActivityWithSelector.cs

```diff
@@ -0,0 +1,85 @@
+using Elsa.Attributes;
+using Elsa.Design;
+using Elsa.Expressions;
+using HtmlAgilityPack;
+using Microsoft.CodeAnalysis.CSharp.Scripting;
+using Microsoft.CodeAnalysis.Scripting;
+using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+// ReSharper disable once CheckNamespace
+namespace Elsa.Activities.Rpa.Web
+{
+    public class WebActivityWithSelector : WebActivity
+    {
+        public WebActivityWithSelector(IServiceProvider sp) : base(sp)
+        {
+        }
+        [ActivityInput(
+            UIHint = ActivityInputUIHints.Dropdown,
+            Hint = "The type of selector to be used to identity the element",
+            Options = new[] { SelectorTypes.ByName, SelectorTypes.ById, SelectorTypes.ByCss },
+            SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid }
+        )]
+        public string? SelectorType { get; set; }
+        [ActivityInput(Hint = "The selector value depends on SelectorType")]
+        public string? SelectorValue { get; set; }
+        public int RetryCount { get; set; } = 5;
+        public TimeSpan RetryInterval { get; set; } = TimeSpan.FromMilliseconds(500);
+        public Func<HtmlNode, bool>? AdvancedSelector { get; set; }
+        internal async Task<IWebElement> GetElement(IWebDriver driver)
+        {
+            var els = await GetElements(driver);
+            if (!els.Any())
+                throw new Exception($"no element found matching the given criterias");
+#pragma warning disable CS8603 // Possible null reference return.
```

Comment:
Any reason not to mark the return type as nullable? Like:

```csharp
internal async Task<IWebElement?> GetElement(IWebDriver driver)
```

Then you can get rid of the pragma warning disable statements.